### DISCUSSION TITLE
Book diff?

### DISCUSSION THREAD

**Discussion by shaunc** [bmoscon/cryptofeed#784] [Q&A] [answered]
I'm wondering if this library supports something like the Binance [book diff stream](https://binance-docs.github.io/apidocs/spot/en/#diff-depth-stream) -- but generalized over other exchanges? This seem to be the most universal (though low level) way to efficiently consume order book data.

**Comment by bmoscon** [comment, marked as answer]
yes thats called the book delta and its stored in the book structure returned to the client via the callbacks. The backends also support storing the book delta (and snapshots)

**Reply by shaunc** [reply]
Thanks - great! (In the [readme](https://github.com/bmoscon/cryptofeed#market-data-channels-public) it only lists book depth.)

**Comment by bmoscon** [comment]
thats because its not a data channel you subscribe to. You subscribe to orderbooks and if the exchange provides it, its part of the data object

**Reply by shaunc** [reply]
Very good. Now that I know I can get what I want, I'll dive in and read the docs. :)

My only other concern is added latency -- I can test myself of course; but do you have a feel for the extra time added for your universal websocket feeds vs raw ones?

**Reply by bmoscon** [reply]
hard to say, what is a raw websocket feed doing? just getting the message isnt very useful, the message has to be parsed and the important information extracted. It could be there is something / some way its being done that is not ideal or could be improved, but I'm not aware of any. I don't think there is much being done by the library that doesnt have to be done by any other interface

**Reply by shaunc** [reply]
Good ... will keep in mind. I'm researching for a client who is doing high-speed trading (well -- perhaps ~1-2 sec turnaround), so they are interested in counting the milliseconds. They are only interested in a couple of exchanges to begin with. But obviously your library takes care of a ton of work, and would allow expansion to many more -- just trying to understand any tradeoffs.
